pkg/health/heartbeat/heartbeat.go

```diff
@@ -241,17 +242,33 @@ func UpdateDeltaSnapshotLease(ctx context.Context, logger *logrus.Entry, prevDel
 	}
 
 	renewedLease := deltaSnapLease.DeepCopy()
-	renewedLease.Spec.HolderIdentity = &revisionNumber
 	renewedTime := time.Now()
 	renewedLease.Spec.RenewTime = &metav1.MicroTime{Time: renewedTime}
 
+	//Update revision number only if revisions in existing lease is lower
+	if len(prevDeltaSnapshots) > 0 {
+		deltaSnap := prevDeltaSnapshots[len(prevDeltaSnapshots)-1]
+		var deltaRev int64
+		if deltaSnapLease.Spec.HolderIdentity != nil {
+			deltaRev, err = strconv.ParseInt(*deltaSnapLease.Spec.HolderIdentity, 10, 64)
+			if err != nil {
+				return err
+			}
+		}
+		if deltaSnapLease.Spec.HolderIdentity == nil || deltaRev < deltaSnap.LastRevision {
```

Comment:
`deltaSnapLease.Spec.HolderIdentity == nil`
This condition is present inside `len(prevDeltaSnapshots)>0{....}` and if `len(prevDeltaSnapshots)>0` which means DeltaSnaposhots present then `deltaSnapLease.Spec.HolderIdentity` can't be nil. right ?

Reply:
The only time `HolderIdentity` can be nil when `len(prevDeltaSnapshots)>0`, is for a fresh cluster when the first delta snap is uploaded and we update the delta lease for the first time
`deltaSnapLease.Spec.HolderIdentity == nil` was meant for this condition

Reply:
hmm, make sense thanks